rotkehlchen/chain/ethereum/modules/weth/decoder.py

```diff
@@ -0,0 +1,136 @@
+            return self._decode_deposit_event(
+                tx_log=tx_log,
+                transaction=transaction,
+                decoded_events=decoded_events,
+                all_logs=all_logs,
+                action_items=action_items,
+            )
+
+        if tx_log.topics[0] == WETH_WITHDRAW_TOPIC:
+            return self._decode_withdrawal_event(
+                tx_log=tx_log,
+                transaction=transaction,
+                decoded_events=decoded_events,
+                all_logs=all_logs,
+                action_items=action_items,
+            )
+
+        return None, None
+
+    def _decode_deposit_event(
+            self,
+            tx_log: EthereumTxReceiptLog,
+            transaction: EvmTransaction,  # pylint: disable=unused-argument
+            decoded_events: List[HistoryBaseEntry],
+            all_logs: List[EthereumTxReceiptLog],  # pylint: disable=unused-argument
+            action_items: Optional[List[ActionItem]],  # pylint: disable=unused-argument
+    ) -> Tuple[Optional[HistoryBaseEntry], Optional[ActionItem]]:
+        depositor = hex_or_bytes_to_address(tx_log.topics[1])
+        deposited_amount_raw = hex_or_bytes_to_int(tx_log.data[:32])
+        deposited_amount = asset_normalized_value(amount=deposited_amount_raw, asset=self.eth)
+
+        for event in decoded_events:
+            if event.event_type == HistoryEventType.SPEND and event.location_label == depositor and event.balance.amount == deposited_amount and event.asset == self.eth:  # noqa: E501
+                event.event_type = HistoryEventType.DEPOSIT
+                event.counterparty = CPT_WETH
+                event.event_subtype = HistoryEventSubType.RECEIVE_WRAPPED
+                event.notes = f'Wrap {deposited_amount} {self.eth.symbol} into {deposited_amount} {self.weth.symbol}'  # noqa: E501
+
+        return None, None
+
+    def _decode_withdrawal_event(
+            self,
+            tx_log: EthereumTxReceiptLog,
+            transaction: EvmTransaction,
+            decoded_events: List[HistoryBaseEntry],
+            all_logs: List[EthereumTxReceiptLog],  # pylint: disable=unused-argument
+            action_items: Optional[List[ActionItem]],  # pylint: disable=unused-argument
+    ) -> Tuple[Optional[HistoryBaseEntry], Optional[ActionItem]]:
+        withdrawer = hex_or_bytes_to_address(tx_log.topics[1])
+        withdrawed_amount_raw = hex_or_bytes_to_int(tx_log.data[:32])
+        withdrawed_amount = asset_normalized_value(amount=withdrawed_amount_raw, asset=self.eth)
+
+        eth_transfer_event = None
+        for event in decoded_events:
+            if event.event_type == HistoryEventType.RECEIVE and event.counterparty == withdrawer and event.balance.amount == withdrawed_amount and event.asset == self.eth:  # noqa: E501
+                # change the sequence index as unwrapping should happen before ETH transfer
+                eth_transfer_event = event
+                break
```

Comment:
For the receive you are already receiving the eth. So you can modify the eth receive event. Here you have duplicated entries 
![image](https://user-images.githubusercontent.com/5068010/197159131-f727b144-f965-40d8-9c36-271f049359a9.png). Just edit the eth event to add the information that is an unwrap